src/Piral.Blazor.Core/ModuleContainerService.cs

```diff
@@ -34,32 +34,8 @@ private IServiceProvider ConfigureLocal(Assembly assembly)
                 ?.GetMethod("ConfigureServices", BindingFlags.Public | BindingFlags.Static, null, new[] { typeof(IServiceCollection) }, null);
 
             configure?.Invoke(null, new[] { sc });
-            return sc.BuildServiceProvider();
-        }
-
-        private IServiceProvider ConfigureGlobal(Assembly assembly)

```

Comment:
What's the story with the removal here?

Right now we take these `Module` classes (just like a `Program` class) and then look for the `ConfigureShared` (among others).

Is this something I missed here? What's the replacement?

Reply:
Maybe I didn't understand the purpose of the ConfigureShared method correctly. According to my intuition, this method would register global dependencies that can be used by all modules? Wouldn't this contradict the principle of micro frontends, as it would introduce dependencies between them?

Reply:
> According to my intuition, this method would register global dependencies that can be used by all modules? Wouldn't this contradict the principle of micro frontends, as it would introduce dependencies between them?

That's up to the user to decide. Blazor itself already runs globally and cannot be isolated - and many dependencies in the Blazor / .NET world also assume a global registration. So this general discussion is a bit pointless.

Reply:
> So this general discussion is a bit pointless.

I completely agree with you. 

The biggest problem I see in implementing a global ServiceProvider is that a ServiceProvider cannot be trivially extended at runtime. For example when module A registers its global services then a ServiceProvider is created and module B can't affect it anymore. I will gladly look at the problem with the global ServiceProvider again in detail. For all I care, we can also make a short call, then I can hopefully convey my thoughts to you a bit more clearly.

Reply:
Yeah sure - but right now I don't see how those two concepts are exclusive. Because (and that is what we had beforehand) you can have a global SP and a local one. If you make every SP local then some things will just not work from the .NET ecosystem.

In my opinion you'll need to have both and the local one should have everything from the global one.

What we can do right now is to keep the `ConfigureShared` but put it all into the local one. This should be good enough.

The last thing I want is a breaking change - so please arrange things in a way that existing implementations would keep on working. Thanks a lot! :beers:

Reply:
I have added the ConfigureShared method again. Furthermore I extended the logic so that when a pilet registers global services, the service providers of all other pilets are updated. With this I wanted to make the registration of dependencies commutative.  I moved the UnitTestes to a separate folder tests and also added a SmokeTest for the DependencyInjection. If you want me to move this somewhere else I can do that. If my changes fit so far, I would like to add several more tests.

Reply:
> If my changes fit so far, I would like to add several more tests.

Yeah these are kick ass - awesome job! Fully appreciated!

>  I moved the UnitTestes to a separate folder tests and also added a SmokeTest for the DependencyInjection.

Outstanding! Really like the structure and love that you put in some smoke tests. This is great!

Reply:
> I have added the ConfigureShared method again. Furthermore I extended the logic so that when a pilet registers global services, the service providers of all other pilets are updated. With this I wanted to make the registration of dependencies commutative.

Sounds reasonable to me - good thinking there. I think this is the right direction. Gold!